# Proyecto final: Análisis conjunto
### Jesús Manuel España Tzec

In [2]:
library(dplyr)
library(tidyr)
library(tidyverse)

Warning message:
"package 'dplyr' was built under R version 3.5.3"
Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union

Warning message:
"package 'tidyverse' was built under R version 3.5.3"Warning message in as.POSIXlt.POSIXct(Sys.time()):
"unable to identify current timezone 'H':
please set environment variable 'TZ'"-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
v ggplot2 3.2.1     v purrr   0.2.5
v tibble  2.1.3     v stringr 1.3.1
v readr   1.3.1     v forcats 0.4.0
Warning message:
"package 'ggplot2' was built under R version 3.5.3"Warning message:
"package 'tibble' was built under R version 3.5.3"Warning message:
"package 'readr' was built under R version 3.5.3"Warning message:
"package 'forcats' was built under R version 3.5.3"-- Conflicts ------------------------------------------ tidyverse_conflicts() 

## 1 Aplica un análisis conjunto adecuado para modelar las preferencias de cada consumidor.
Primero cargamos el conjunto de datos que vamos a utilizar, *Device Full Data* es un archivo que contiene la calificación de diez consumidores sobre dieciocho productos.

In [3]:
dat <- read.csv("device_fullData.csv")
head(dat)

Rating_1 Rating_2 Rating_3 Rating_4 Rating_5 Rating_6 Rating_7 Rating_8
1 3        4        3        1        2        4        5        2       
2 1        2        2        2        3        1        1        1       
3 3        2        4        4        2        3        5        2       
4 2        3        5        1        3        3        3        1       
5 4        5        2        4        4        4        5        5       
6 3        1        2        1        3        1        1        2       
  Rating_9 Rating_10 Price BrandOS Storage ImageQuality
1 3        1          40   Roku    0GB     HDR+4K      
2 2        2         100   Google  6GB     HD          
3 4        4          70   Amazon  6GB     HDR+4K      
4 2        1          70   Roku    0GB     HD          
5 4        4          40   Google  16GB    HDR+4K      
6 4        1         100   Apple   0GB     HDR+4K

Creamos una función para obtener la calificación de cada usuario para cada producto.

In [4]:
user <- function(dat, from, co, tam = 5 ){
    exc <- colnames(dat[co])
    val <- ncol(dat)
    while (val > tam){
        if(exc != colnames(dat[from])){
            dat <- dat[-from]
            val <- ncol(dat)
        }
        else{
            from <- from + 1
        }
    }
    return(dat)
}

Cambiamos el tipo de la columna *Price* para que pueda ser manejable y creamo el modelo lineal para el primer usuario. 

In [5]:
dat$Price <-as.factor(dat$Price)
dat1 <- user(dat, 1, 1)

reg1 <- lm(Rating_1~., data = dat1)
summary(reg1)


Call:
lm(formula = Rating_1 ~ ., data = dat1)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.4919 -0.1422 -0.0074  0.1843  0.3515 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)         2.52471    0.32073   7.872 0.000101 ***
Price70            -0.65106    0.21628  -3.010 0.019654 *  
Price100           -0.96429    0.24395  -3.953 0.005512 ** 
BrandOSApple        0.96081    0.25456   3.774 0.006943 ** 
BrandOSGoogle       0.04808    0.26851   0.179 0.862971    
BrandOSRoku         0.42510    0.25456   1.670 0.138860    
Storage16GB         0.57490    0.25456   2.258 0.058469 .  
Storage32GB         0.87298    0.25456   3.429 0.010995 *  
Storage6GB          0.28571    0.24395   1.171 0.279830    
ImageQualityHD     -0.65021    0.23604  -2.755 0.028314 *  
ImageQualityHDR+4K  0.54210    0.23604   2.297 0.055265 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.3726 on 7 degrees of freedom


Con el modelo del primer usuario podemos explicar el 91% de la información. Los factores que pueden explicarse como una combinación lineal como:
***-0.65106(Price70) - 0.96429(Price100) + 0.96081(BrandOSApple) + 0.87298(Storage32GB) - 0.65021(ImageQualityHD)***.

Creamos el modelo para el segundo usuario.

In [6]:
dat2 <- user(dat, 1, 2)
  
reg2 <- lm(Rating_2~., data = dat2)
summary(reg2)


Call:
lm(formula = Rating_2 ~ ., data = dat2)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.40897 -0.33108  0.00757  0.26114  0.50520 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1.0196     0.4168   2.446 0.044364 *  
Price70             -0.7645     0.2811  -2.720 0.029779 *  
Price100            -1.2857     0.3170  -4.055 0.004838 ** 
BrandOSApple        -0.4443     0.3308  -1.343 0.221193    
BrandOSGoogle        1.1538     0.3490   3.306 0.013006 *  
BrandOSRoku          1.8414     0.3308   5.566 0.000846 ***
Storage16GB          1.1586     0.3308   3.502 0.009967 ** 
Storage32GB          1.3124     0.3308   3.967 0.005413 ** 
Storage6GB           0.7143     0.3170   2.253 0.058943 .  
ImageQualityHD       0.7734     0.3068   2.521 0.039751 *  
ImageQualityHDR+4K   1.3888     0.3068   4.527 0.002709 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.4843 on 7 degrees o

Para el segundo modelo se puede explicar el 94% de la información. Los factores que pueden explicarse como una combinación lineal como: ***- 0.7645(Price70) - 1.2857(Price100 ) + 1.1538(BrandOSGoogle) + 1.8414(BrandOSRoku) + 1.1586(Storage16GB) + 1.3124(Storage32GB) + 0.7734(ImageQualityHD) + 1.3888(ImageQualityHDR+4K)***.

Creamos el modelo para el tercer usuario.

In [7]:
dat3 <- user(dat, 1, 3)
  
reg3 <- lm(Rating_3~., data = dat3)
summary(reg3)


Call:
lm(formula = Rating_3 ~ ., data = dat3)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.52799 -0.16814 -0.04733  0.16695  0.46926 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)         4.70382    0.36347  12.942 3.82e-06 ***
Price70             0.05116    0.24510   0.209 0.840607    
Price100           -1.35714    0.27645  -4.909 0.001735 ** 
BrandOSApple       -0.36984    0.28848  -1.282 0.240660    
BrandOSGoogle      -1.92308    0.30429  -6.320 0.000397 ***
BrandOSRoku        -0.01270    0.28848  -0.044 0.966124    
Storage16GB         0.76270    0.28848   2.644 0.033238 *  
Storage32GB         1.83962    0.28848   6.377 0.000375 ***
Storage6GB          0.14286    0.27645   0.517 0.621251    
ImageQualityHD      0.35967    0.26749   1.345 0.220698    
ImageQualityHDR+4K -1.33264    0.26749  -4.982 0.001597 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.4223 on 7 degrees o

Para el tercer modelo se puede explicar el 96% de la información. Los factores que pueden explicarse como una combinación lineal como: ***- 1.35714(Price100) - 1.92308(BrandOSGoogle) + 0.76270(Storage16GB) + 1.83962(Storage32GB) - 1.33264(ImageQualityHDR+4K)***.

Creamos el modelo para el cuarto usuario.

In [8]:
dat4 <- user(dat, 1, 4)
  
reg4 <- lm(Rating_4~., data = dat4)
summary(reg4)


Call:
lm(formula = Rating_4 ~ ., data = dat4)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.35907 -0.13079  0.03089  0.17664  0.29730 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)          3.9855     0.2731  14.595 1.69e-06 ***
Price70             -0.4353     0.1841  -2.364 0.050044 .  
Price100            -1.3571     0.2077  -6.534 0.000324 ***
BrandOSApple        -1.6718     0.2167  -7.713 0.000115 ***
BrandOSGoogle       -1.2500     0.2286  -5.468 0.000938 ***
BrandOSRoku         -2.5647     0.2167 -11.833 6.99e-06 ***
Storage16GB          1.3147     0.2167   6.066 0.000508 ***
Storage32GB          1.5647     0.2167   7.219 0.000175 ***
Storage6GB           0.8929     0.2077   4.299 0.003572 ** 
ImageQualityHD      -0.1622     0.2010  -0.807 0.446278    
ImageQualityHDR+4K  -0.1622     0.2010  -0.807 0.446278    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.3173 on 7 degrees o

Para el cuarto modelo se puede explicar el 97% de la información. Los factores que pueden explicarse como una combinación lineal como: ***- 1.3571(Price100) - 1.6718(BrandOSApple) - 1.2500(BrandOSGoogle) - 2.5647(BrandOSRoku) + 1.3147(Storage16GB) + 1.5647(Storage32GB) + 0.8929(Storage6GB)***.

Creamos el modelo para el quinto usuario.

In [9]:
dat5 <- user(dat, 1, 5)
  
reg5 <- lm(Rating_5~., data=dat5)
summary(reg5)


Call:
lm(formula = Rating_5 ~ ., data = dat5)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.37526 -0.22531 -0.02075  0.17770  0.49707 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1.9079     0.3252   5.866 0.000620 ***
Price70             -0.4725     0.2193  -2.154 0.068169 .  
Price100            -1.6071     0.2474  -6.497 0.000335 ***
BrandOSApple         3.1490     0.2581  12.199 5.69e-06 ***
BrandOSGoogle        1.5673     0.2723   5.756 0.000694 ***
BrandOSRoku          0.7561     0.2581   2.929 0.022047 *  
Storage16GB          0.7439     0.2581   2.882 0.023595 *  
Storage32GB          1.0612     0.2581   4.111 0.004511 ** 
Storage6GB           0.6429     0.2474   2.599 0.035488 *  
ImageQualityHD       0.3113     0.2394   1.301 0.234537    
ImageQualityHDR+4K  -0.4194     0.2394  -1.752 0.123165    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.3779 on 7 degrees o

Para el quinto modelo se puede explicar el 97% de la información. Los factores que pueden explicarse como una combinación lineal como: ***- 1.6071(Price100) + 3.1490(BrandOSApple) + 1.5673(BrandOSGoogle) + 0.7561(BrandOSRoku) + 0.7439(Storage16GB) + 1.0612(Storage32GB) + 0.6429(Storage6GB)***.

Creamos el modelo para el sexto usuario.

In [10]:
dat6 <- user(dat, 1, 6)
reg6 <- lm(Rating_6~., data=dat6)
summary(reg6)


Call:
lm(formula = Rating_6 ~ ., data = dat6)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.55257 -0.20884 -0.00494  0.18384  0.53464 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)          2.9632     0.3980   7.445 0.000144 ***
Price70             -1.4402     0.2684  -5.366 0.001046 ** 
Price100            -2.2857     0.3027  -7.551 0.000132 ***
BrandOSApple         0.1256     0.3159   0.398 0.702784    
BrandOSGoogle       -1.0577     0.3332  -3.174 0.015614 *  
BrandOSRoku          0.4113     0.3159   1.302 0.234108    
Storage16GB          1.3387     0.3159   4.238 0.003851 ** 
Storage32GB          1.7810     0.3159   5.638 0.000784 ***
Storage6GB           0.7143     0.3027   2.360 0.050376 .  
ImageQualityHD       0.9802     0.2929   3.347 0.012309 *  
ImageQualityHDR+4K   0.7495     0.2929   2.559 0.037623 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.4624 on 7 degrees o

Para el sexto modelo se puede explicar el 94% de la información. Los factores que pueden explicarse como una combinación lineal como: ****- 1.4402(Price70) - 2.2857(Price100) - 1.0577(BrandOSGoogle) + 1.3387(Storage16GB) + 1.7810(Storage32GB) + 0.9802(ImageQualityHD) + 0.7495(ImageQualityHDR+4K)***.

Creamos el modelo para el septimo usuario.

In [11]:
dat7 <- user(dat, 1, 7)
  
reg7 <- lm(Rating_7~., data=dat7)
summary(reg7)


Call:
lm(formula = Rating_7 ~ ., data = dat7)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.25733 -0.04570  0.02079  0.08826  0.30795 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)          3.5962     0.1943  18.512 3.33e-07 ***
Price70             -0.5632     0.1310  -4.299 0.003569 ** 
Price100            -2.4643     0.1478 -16.678 6.81e-07 ***
BrandOSApple        -1.6174     0.1542 -10.490 1.56e-05 ***
BrandOSGoogle       -1.1827     0.1626  -7.272 0.000167 ***
BrandOSRoku         -0.4031     0.1542  -2.614 0.034687 *  
Storage16GB          0.9031     0.1542   5.857 0.000626 ***
Storage32GB          0.9704     0.1542   6.294 0.000407 ***
Storage6GB           0.5357     0.1478   3.626 0.008447 ** 
ImageQualityHD       0.4194     0.1430   2.934 0.021908 *  
ImageQualityHDR+4K   1.6887     0.1430  11.811 7.07e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.2257 on 7 degrees o

Para el septimo modelo se puede explicar el 99% de la información. Los factores que pueden explicarse como una combinación lineal como: ***- 0.5632(Price70) - 2.4643(Price100) - 1.6174(BrandOSApple) - 1.1827(BrandOSGoogle) - 0.4031(BrandOSRoku) + 0.9031(Storage16GB) + 0.9704(Storage32GB) + 0.5357(Storage6GB) + 0.4194(ImageQualityHD) + 1.6887(ImageQualityHDR+4K)***.

Creamos el modelo para el octavo usuario.

In [12]:
dat8 <- user(dat, 1, 8)
  
reg8 <- lm(Rating_8~., data=dat8)
summary(reg8)


Call:
lm(formula = Rating_8 ~ ., data = dat8)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.65039 -0.24348 -0.05275  0.23900  0.52807 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1.9237     0.4649   4.138 0.004362 ** 
Price70             -0.8996     0.3135  -2.869 0.024014 *  
Price100            -2.2857     0.3536  -6.464 0.000346 ***
BrandOSApple         1.6043     0.3690   4.347 0.003365 ** 
BrandOSGoogle        1.2308     0.3892   3.162 0.015884 *  
BrandOSRoku         -0.3600     0.3690  -0.976 0.361779    
Storage16GB          1.1100     0.3690   3.008 0.019718 *  
Storage32GB          1.3408     0.3690   3.633 0.008360 ** 
Storage6GB           0.4643     0.3536   1.313 0.230613    
ImageQualityHD      -0.1237     0.3422  -0.362 0.728379    
ImageQualityHDR+4K   0.7994     0.3422   2.336 0.052137 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.5402 on 7 degrees o

Para el octavo modelo se puede explicar el 93% de la información. Los factores que pueden explicarse como una combinación lineal como: ***- 0.8996(Price70) - 2.2857(Price100) + 1.6043(BrandOSApple) + 1.2308(BrandOSGoogle) + 1.1100(Storage16GB) + 1.3408(Storage32GB)***.

Creamos el modelo para el noveno usuario.

In [13]:
dat9 <- user(dat, 1, 9)
  
reg9 <- lm(Rating_9~., data=dat9)
summary(reg9)


Call:
lm(formula = Rating_9 ~ ., data = dat9)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.34200 -0.11238 -0.06357  0.11946  0.38697 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1.6191     0.2527   6.408 0.000365 ***
Price70             -0.2080     0.1704  -1.221 0.261677    
Price100            -0.3214     0.1922  -1.673 0.138343    
BrandOSApple         0.6125     0.2005   3.054 0.018469 *  
BrandOSGoogle       -0.7981     0.2115  -3.773 0.006958 ** 
BrandOSRoku         -0.8160     0.2005  -4.069 0.004755 ** 
Storage16GB          0.8160     0.2005   4.069 0.004755 ** 
Storage32GB          1.2679     0.2005   6.322 0.000396 ***
Storage6GB           0.1786     0.1922   0.929 0.383723    
ImageQualityHD       1.4610     0.1860   7.857 0.000102 ***
ImageQualityHDR+4K   2.2687     0.1860  12.200 5.69e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.2936 on 7 degrees o

Para el noveno modelo se puede explicar el 97% de la información. Los factores que pueden explicarse como una combinación lineal como: ***0.6125(BrandOSApple) - 0.7981(BrandOSGoogle) - 0.8160(BrandOSRoku) + 0.8160(Storage16GB) + 1.2679(Storage32GB) + 1.4610(ImageQualityHD) + 2.2687(ImageQualityHDR+4K)***.

Creamos el modelo para el decimo usuario.

In [14]:
dat10 <- user(dat, 1, 10)
  
reg10 <- lm(Rating_10~., data=dat10)
summary(reg10)


Call:
lm(formula = Rating_10 ~ ., data = dat10)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.40700 -0.08586  0.04672  0.12868  0.30940 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)          2.9111     0.2578  11.291 9.56e-06 ***
Price70             -0.6573     0.1739  -3.781 0.006884 ** 
Price100            -1.0714     0.1961  -5.464 0.000942 ***
BrandOSApple        -1.9429     0.2046  -9.495 3.01e-05 ***
BrandOSGoogle       -1.3077     0.2158  -6.059 0.000512 ***
BrandOSRoku         -2.6215     0.2046 -12.811 4.09e-06 ***
Storage16GB          1.1215     0.2046   5.481 0.000925 ***
Storage32GB          1.8138     0.2046   8.864 4.71e-05 ***
Storage6GB           0.6786     0.1961   3.460 0.010542 *  
ImageQualityHD       1.1965     0.1897   6.306 0.000402 ***
ImageQualityHDR+4K   0.9657     0.1897   5.090 0.001416 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.2995 on 7 degrees

Para el decimo modelo se puede explicar el 97% de la información. Los factores que pueden explicarse como una combinación lineal como: ***- 0.6573(Price70) - 1.0714(Price100) - 1.9429(BrandOSApple) - 1.3077(BrandOSGoogle) - 2.6215(BrandOSRoku) + 1.1215(Storage16GB) + 1.8138(Storage32GB) + 0.6786(Storage6GB) + 1.1965(ImageQualityHD) + 0.9657(ImageQualityHDR+4K)***.

## 2. Mediante un modelo global, ordena los atributos de acuerdo a la importancia que le otorgan los consumidores.

In [15]:
data <- dat %>% gather(Rating_1,Rating_2,Rating_3,Rating_4,Rating_5,Rating_6,Rating_7,Rating_8,Rating_9,Rating_10)
data <- data %>% dplyr::rename(Rating_3 = Rating_1, Rating_4 = Rating_2)

data2 <- user(dat, 4, 4, 8)
data2 <- data2 %>% gather(Rating_3,Rating_4,Rating_1,Rating_2)

Hacemos el modelo global con los dataframes anteriores ya ordenados.

In [16]:
data3 <-rbind(data,data2)
data3 <- data3[-5]
reg11 <- lm(Rating_4~., data = data3)
summary(reg11)


Call:
lm(formula = Rating_4 ~ ., data = data3)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.2180 -0.9049  0.0032  0.8280  3.2364 

Coefficients:
                   Estimate Std. Error t value Pr(>|t|)    
(Intercept)         2.71549    0.31498   8.621 4.60e-15 ***
Price70            -0.60405    0.21240  -2.844  0.00501 ** 
Price100           -1.50000    0.23957  -6.261 3.05e-09 ***
BrandOSApple        0.04059    0.25000   0.162  0.87121    
BrandOSGoogle      -0.35192    0.26370  -1.335  0.18382    
BrandOSRoku        -0.33441    0.25000  -1.338  0.18281    
Storage16GB         0.98441    0.25000   3.938  0.00012 ***
Storage32GB         1.38248    0.25000   5.530 1.20e-07 ***
Storage6GB          0.52500    0.23957   2.191  0.02979 *  
ImageQualityHD      0.45655    0.23181   1.969  0.05053 .  
ImageQualityHDR+4K  0.64886    0.23181   2.799  0.00572 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.157 on 169 degrees of freedo

El modelo permite explicar 32% de la información, siendo los predictores más significativos son *Price70, Price100, Storage16GB, Storage32GB, Storage6GB, ImageQualityHDR+4K*.
La combinación lineal queda como: ***- 0.60405(Price70) - 1.50000(Price100) + 0.98441(Storage16GB) + 1.38248(Storage32GB) + 0.52500(Storage6GB) + 0.64886(ImageQualityHDR+4K)***.

##  3.Utilizando los modelos que construiste en el punto 1, identifica cuál de las opciones de la línea de producción de Google tendrá la mayor aceptación entre los consumidores (más consumidores lo eligirán). Observa que el nuevo producto deberá competir con los tres que se ofrecen actualmente.

Cargamos el nuevo conjunto de datos y observamos las primeras entradas.

In [17]:
data2 <- read.csv("device_Products.csv")
head(data2)

Price BrandOS Storage ImageQuality Available
1  70   Roku    6GB     HDR+4K       1        
2  70   Amazon  16GB    Full HD      1        
3 100   Apple   32GB    HDR+4K       1        
4  40   Google  6GB     HD           0        
5  70   Google  32GB    Full HD      0        
6  70   Google  16GB    HDR+4K       0

Cambiamos el tipo de la columna *Price* y realizamos las predicciones para cada usuario.

In [18]:
data2$Price <-as.factor(data2$Price)
pred1 <- predict(reg1, data2)
pred2 <- predict(reg2, data2)
pred3 <- predict(reg3, data2)
pred4 <- predict(reg4, data2)
pred5 <- predict(reg5, data2)
pred6 <- predict(reg6, data2)
pred7 <- predict(reg7, data2)
pred8 <- predict(reg8, data2)
pred9 <- predict(reg9, data2)
pred10 <- predict(reg10, data2)

Agrupamos todas las predicciones en un dataframe

In [19]:
dpred <- data.frame(
    "Apple" = c(pred1[3], pred2[3], pred3[3], pred4[3], pred5[3], pred6[3], pred7[3], pred8[3], pred9[3], pred10[3]),
    "Amazon" = c(pred1[2], pred2[2], pred3[2], pred4[2], pred5[2], pred6[2], pred7[2], pred8[2], pred9[2], pred10[2]),
    "Roku" = c(pred1[1], pred2[1], pred3[1], pred4[1], pred5[1], pred6[1], pred7[1], pred8[1], pred9[1], pred10[1]), 
    "Google_6GB_HD" = c(pred1[4], pred2[4], pred3[4], pred4[4], pred5[4], pred6[4], pred7[4], pred8[4], pred9[4], pred10[4]),
    "Google_32GB_FullHD" = c(pred1[5], pred2[5], pred3[5], pred4[5], pred5[5], pred6[5], pred7[5], pred8[5], pred9[5], pred10[5]),
    "Google_16GB_HDR4K" = c(pred1[6], pred2[6], pred3[6], pred4[6], pred5[6], pred6[6], pred7[6], pred8[6], pred9[6], pred10[6])
)
dpred 

Apple    Amazon   Roku     Google_6GB_HD Google_32GB_FullHD
1  3.936312 2.448545 3.126559 2.208290      2.794699          
2  1.990793 1.413721 4.199584 3.661123      2.721414          
3  3.483813 5.517672 3.552495 3.283264      4.671518          
4  2.359073 4.864865 1.716216 3.466216      3.864865          
5  4.091532 2.179314 2.415021 4.429444      4.063929          
6  3.333568 2.861746 3.398129 3.600052      2.246362          
7  2.173578 3.936071 4.854210 3.368633      2.820686          
8  3.382425 2.134096 1.927755 3.495062      3.595634          
9  5.446893 2.227131 3.042360 2.460629      1.880977          
10 2.676233 3.375260 1.276507 3.478430      2.759875          
   Google_16GB_HDR4K
1  3.038721         
2  3.956341         
3  2.261954         
4  3.452703         
5  3.327183         
6  2.553534         
7  4.442048         
8  4.164241         
9  3.697765         
10 3.033264

Los resultados que tenemos es que para el usuario 1 el producto que tiene más posibilidad de ser adquirido es el de Apple de 32GB y calidad HDR+4K, para el usuario 2 el producto a escoger es el de Google con 16GB y HDR+4K, para el usuario 3 es el de Amazon con 16GB y FullHD, para el usuario 4 es igual el de Amazon, para el usuario 5 es el producto de Google con 6GB y HD, para el usuario 6 es el mismo que el anterior usuario, para el usuario 7 el producto con mayor aceptación es el de Roku con 6GB y HDR+4K, para el usuario 8 el producto a escoger es el de Google con 16GB con HDR+4K, para el usuario 9 el producto de selección es el de Apple de 32GB y HDR+4K, para el usuario 10 el producto con más posibilidad de ser escogido es el de Google con 6GB y calidad HD.

## 4. Identifica con cuál de los productos se obtendrán la mayores ganancias. Observa que no necesariamente será el más aceptado por los usuarios.

Cargamos el nuevo de conjunto de datos y observamos los primeros valores.

In [20]:
data3 <- read.csv("device_ProductLine.csv")
head(data3)

Price BrandOS Storage ImageQuality Margin
1 40    Google  0GB     HD           25    
2 40    Google  6GB     HD           17    
3 40    Google  16GB    HD           10    
4 40    Google  32GB    HD            5    
5 40    Google  0GB     Full HD      15    
6 40    Google  0GB     HDR+4K        5

Realizamos el modelo con los dato nuevos.

In [21]:
data3$Price <-as.factor(data3$Price)

reg12 <- lm(Margin~Price+Storage+ImageQuality, data=data3)
summary(reg12)

Warning message in summary.lm(reg12):
"essentially perfect fit: summary may be unreliable"


Call:
lm(formula = Margin ~ Price + Storage + ImageQuality, data = data3)

Residuals:
       Min         1Q     Median         3Q        Max 
-2.897e-14 -3.043e-15  1.866e-16  4.038e-15  2.214e-14 

Coefficients:
                     Estimate Std. Error    t value Pr(>|t|)    
(Intercept)         1.500e+01  6.551e-15  2.290e+15   <2e-16 ***
Price70             3.000e+01  6.213e-15  4.829e+15   <2e-16 ***
Price100            6.000e+01  6.153e-15  9.751e+15   <2e-16 ***
Storage16GB        -1.500e+01  5.941e-15 -2.525e+15   <2e-16 ***
Storage32GB        -2.000e+01  6.717e-15 -2.977e+15   <2e-16 ***
Storage6GB         -8.000e+00  5.941e-15 -1.347e+15   <2e-16 ***
ImageQualityHD      1.000e+01  5.496e-15  1.819e+15   <2e-16 ***
ImageQualityHDR+4K -1.000e+01  5.757e-15 -1.737e+15   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.151e-14 on 20 degrees of freedom
Multiple R-squared:      1,	Adjusted R-squared:      1 
F-statistic: 1.50

Realizamos la predicción con este conjunto de datos para observar cual es el mejor. 

In [22]:
predict(reg12,data2[4:6,])

4  5  6 
17 25 20

Los resultados se leen de arriba hacia abajo siendo el primero Google con 6GB, el segundo e Google 32GB y el último es Google 16GB. El modelo señala que el de mayor ganancia es el modelo de 32GB y el de menor ganancia es para el modelo de 6GB.